agent/grpc-external/services/resource/server.go

```diff
@@ -58,3 +56,31 @@ func (s *Server) Watch(req *pbresource.WatchRequest, ws pbresource.ResourceServi
 	// TODO
 	return nil
 }
+
+func (s *Server) resolveType(typ *pbresource.Type) (*resource.Registration, error) {
+	v, ok := s.registry.Resolve(typ)
+	if ok {
+		return &v, nil
+	}
+	return nil, status.Errorf(
+		codes.InvalidArgument,
+		"resource type %s not registered", resource.ToGVK(typ),
+	)
+}
+
+func readConsistencyFrom(ctx context.Context) storage.ReadConsistency {
```

Comment:
This will replace `isConsistentRead(...)` in the branches for the Read and WatchList endpoints once this is merged.

Reply:
Nice! Sounds good 👍🏻

Reply:
Fixed in https://github.com/hashicorp/consul/pull/16753/commits/9965d528a03ca866b02f4c5a82a55deb7b1e57eb